src/main/java/com/google/growpod/servlets/UserServlet.java

```diff
@@ -172,4 +176,33 @@ private User getUserById(String id) {
     }
     return USER_GARDEN_ADMIN_LIST_MAP.get(id);
   }
+
+  /**
+   * Currently this Post function will get the user data in the form of a Json object,turn the Json
+   * object into a User object and put that information into a map.
+   *
+   * @param req Information about the POST request
+   * @param resp Information about the servlet's response
+   */
+  @Override
+  public void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    Gson gson = new Gson();
+    resp.setContentType("application/json");
+    String jsonString = req.getParameter("userData");
+    User userData = gson.fromJson(jsonString, User.class);
+    addToMap(userData);
+  }
+
+  /**
+   * This function puts the user into userMap.
+   *
+   * @param user user object holding user data: id,email,name,bio,zip. The runtime of this function
+   *     would be O(1) but a worse case senareo would be O(n) with n being the elements in User
```

Comment:
Nit: A shorthand for this comment would be just `expected O(1) time`.

Reply:
nit: senareo --> scenario